src/codegen/codegen.cpp

```diff
@@ -92,9 +92,9 @@ llvm::Value *CodeGen::AllocateVariable(llvm::Type *type,
 
   auto *entry_block = code_context_.GetCurrentFunction()->GetEntryBlock();
   if (entry_block->empty()) {
-    return new llvm::AllocaInst(type, name, entry_block);
+    return new llvm::AllocaInst(type, 0, name, entry_block);
```

Comment:
LLVM 5.0+ added a parameter argument to the constructor for this class, so we're actually calling different versions of the constructor when using pre-5.0 and 5.0+. To ensure consistency, we should add version checks around these calls using the `LLVM_VERSION_GE` macros (in `macros.h`).

Reply:
So if LLVM_VERSION_GE(5, 0) use the new calls, otherwise use the old calls?